File: app/src/main/java/com/booboot/vndbandroid/model/vndb/Results.java
package com.booboot.vndbandroid.model.vndb;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

/**
 * Created by od on 12/03/2016.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class Results<T> {
    private int num;
    private boolean more;
    private List<T> items;

    public int getNum() {
        return num;
    }

    public void setNum(int num) {
        this.num = num;
    }

    public boolean isMore() {
        return more;
    }

    public void setMore(boolean more) {
        this.more = more;
    }

    public List<T> getItems() {
        return items;
    }

    public void setItems(List<T> items) {
        this.items = items;
    }
}

File: app/src/main/java/com/booboot/vndbandroid/model/vndb/Results.kt
package com.booboot.vndbandroid.model.vndb

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Results<T>(
    var num: Int = 0,
    var more: Boolean = false,
    var items: MutableList<T> = mutableListOf()
)